evm-contracts/src/v0.8/dev/ValidatorProxy.sol

```diff
@@ -0,0 +1,251 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./ConfirmedOwner.sol";
+import "../interfaces/AggregatorValidatorInterface.sol";
+
+contract ValidatorProxy is AggregatorValidatorInterface, ConfirmedOwner {
+
+  /// @notice Uses a single storage slot to store the current address
+  struct ProxyConfiguration {
+    address target;
```

Comment:
This isn't ideal because this struct is also used for the aggregator, which doesn't conform to that interface.

Reply:
How about separate structs for separate purposes?

Reply:
We could, but they are essentially the same thing, and it would make other state vars mismatch. So this:
```solidity
  /// @notice Uses a single storage slot to store the current address
  struct ProxyConfiguration {
    address target;
    bool hasNewProposal;
  }

  // Configuration for the current aggregator
  ProxyConfiguration private s_currentAggregator;
  // Proposed aggregator address
  address private s_proposedAggregator;

  // Configuration for the current validator
  ProxyConfiguration private s_currentValidator;
  // Proposed validator address
  address private s_proposedValidator;

  event AggregatorProposed(
    address indexed aggregator
  );
  event AggregatorUpgraded(
    address indexed previous,
    address indexed current
  );
  event ValidatorProposed(
    address indexed validator
  );
  event ValidatorUpgraded(
    address indexed previous,
    address indexed current
  );
  /// @notice The proposed aggregator called validate, but the call was not passed on to any validators
  event ProposedAggregatorValidateCall(
    address indexed proposed,
    uint256 previousRoundId,
    int256 previousAnswer,
    uint256 currentRoundId,
    int256 currentAnswer
  );
```

becomes a bit messier:

```solidity
  /// @notice Uses a single storage slot to store the current address
  struct AggregatorConfiguration {
    address target;
    bool hasNewProposal;
  }

  struct ValidatorConfiguration {
    AggregatorValidatorInterface target;
    bool hasNewProposal;
  }

  // Configuration for the current aggregator
  AggregatorConfiguration private s_currentAggregator;
  // Proposed aggregator address
  address private s_proposedAggregator;

  // Configuration for the current validator
  ValidatorConfiguration private s_currentValidator;
  // Proposed validator address
  AggregatorValidatorInterface private s_proposedValidator;

  event AggregatorProposed(
    address indexed aggregator
  );
  event AggregatorUpgraded(
    address indexed previous,
    address indexed current
  );
  event ValidatorProposed(
    AggregatorValidatorInterface indexed validator
  );
  event ValidatorUpgraded(
    AggregatorValidatorInterface indexed previous,
    AggregatorValidatorInterface indexed current
  );
  /// @notice The proposed aggregator called validate, but the call was not passed on to any validators
  event ProposedAggregatorValidateCall(
    address indexed proposed,
    uint256 previousRoundId,
    int256 previousAnswer,
    uint256 currentRoundId,
    int256 currentAnswer
  );
```

Since aggregators will be the `address` type anyway, it starts to get a bit messy. It's purely aesthetic, but I feel using `address` across the board and having a single struct is more elegant.

Reply:
To me the second looks cleaner plus you get more type safety.